velox/exec/tests/CustomizedJoinBridgeTest.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/exec/JoinBridge.h"
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+
+#include <gmock/gmock.h>
+
+using namespace facebook::velox;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+
+class MockJoinBridge : public JoinBridge {
+ public:
+  bool isMockJoinBridge() {
+    return true;
+  }
+};
+
+class MockJoinNode : public core::PlanNode {
+ public:
+  MockJoinNode(const core::PlanNodeId& id) : PlanNode(id) {}
+
+  MOCK_METHOD(const RowTypePtr&, outputType, (), (const, override));
+  MOCK_METHOD(
+      const std::vector<std::shared_ptr<const PlanNode>>&,
+      sources,
+      (),
+      (const, override));
+  MOCK_METHOD(std::string_view, name, (), (const, override));
+  MOCK_METHOD(
+      void,
+      addDetails,
+      (std::stringstream & stream),
+      (const, override));
+};
+
+class CustomizedJoinBridgeTranslator : public Task::JoinBridgeTranslator {
+  std::shared_ptr<JoinBridge> translate(
+      const std::shared_ptr<const core::PlanNode>& node) {
+    if (auto joinNode = std::dynamic_pointer_cast<const MockJoinNode>(node)) {
+      return std::shared_ptr<JoinBridge>(std::make_shared<MockJoinBridge>());
+    }
+    return nullptr;
+  }
+};
+
+class MockJoinBridgeTest : public OperatorTestBase {
```

Comment:
It is not clear to me what this test is testing. Would you clarify? It would be helpful to create an example of a custom join node and custom join operator and make sure a query plan with such a node runs successfully and returns correct results.

Reply:
Thanks for your review. This UT only target to verify the join bridge translate API. I am not sure whether we need really a custom join node to verify a query plan. But anyway let me take a try.